### PR TITLE
fix: fix custom llama and qwen state dict adapter concat and split

### PR DESCRIPTION
**Pull request by HuiyingLi**
The expected layout is [Q,K,V]. If load/save with _safe_concat/split on local tensor only while when multigpu FSDP shard on dim0 (default), the shards have incorrect q/k/v content (interleaved), causing multigpu fsdp (only) to have high loss, while the single gpu remains correct.

### CODE REVIEW

File: nemo_automodel/components/models/common/combined_projection/state_dict_adapter.py [hunk 1]
```diff
@@ -37,55 +38,58 @@ def _is_dtensor(tensor: torch.Tensor) -> bool:
...
-        if sum(local_split_sizes) != local_size:
-            raise RuntimeError(
-                f"Split size mismatch: local_split_sizes={local_split_sizes} sum to {sum(local_split_sizes)}, "
-                f"but local tensor size is {local_size}. This indicates an incorrect split calculation."
-            )
-
-        local_splits = local_tensor.split(local_split_sizes, dim=dim)
+        try:
+            return list(tensor.split(split_sizes, dim=dim))
+        except Exception:
```
Comment (ZhiyuLi-Nvidia) [inline nemo_automodel/components/models/common/combined_projection/state_dict_adapter.py]: May I know the reason to handle exception here?

File: nemo_automodel/components/models/common/combined_projection/state_dict_adapter.py [hunk 2]
```diff
@@ -37,55 +38,58 @@ def _is_dtensor(tensor: torch.Tensor) -> bool:
...
+            pass
+
+        # Fallback for small 1D Shard(0) tensors (e.g., fused QKV bias)
+        if (
+            tensor.ndim == 1
+            and dim == 0
+            and len(tensor.placements) == 1
+            and isinstance(tensor.placements[0], Shard)
+            and tensor.placements[0].dim == 0
```
Comment (ZhiyuLi-Nvidia) [inline nemo_automodel/components/models/common/combined_projection/state_dict_adapter.py]: Emm there are 5 conditional checks. So there would be all together 2^5 cases, would `else` handle the other cases?

File: tests/unit_tests/distributed/test_parallelizer_utils.py [hunk 3]
```diff
@@ -227,6 +227,163 @@ def fake__fully_shard(mod, *, mesh, mp_policy, offload_policy):
...
+        num_attention_heads=2,
+        num_key_value_heads=2,
+        hidden_size=4,
+        head_dim=2,
+        tie_word_embeddings=False,
+    )
+    adapter = mod.CombinedProjectionStateDictAdapter(cfg)
+
+    hf_sd = {
+        "model.layers.0.mlp.gate_proj.weight": mod.torch.ones((3, 4)),
```
Comment (ZhiyuLi-Nvidia) [inline tests/unit_tests/distributed/test_parallelizer_utils.py]: can we use random values so that make it a more challenging test cases?

We can set random seed for repro as well.

Comment (HuiyingLi) [reply tests/unit_tests/distributed/test_parallelizer_utils.py]: Updated the test to random val, thanks!

**Review by akoumpa — APPROVED** [review]
